sender.go

```diff
@@ -3,18 +3,28 @@ package amqp
 import (
 	"context"
 	"encoding/binary"
+	"errors"
 	"fmt"
 	"sync"
 	"sync/atomic"
 
 	"github.com/Azure/go-amqp/internal/buffer"
+	"github.com/Azure/go-amqp/internal/debug"
 	"github.com/Azure/go-amqp/internal/encoding"
 	"github.com/Azure/go-amqp/internal/frames"
+	"github.com/Azure/go-amqp/internal/shared"
 )
 
 // Sender sends messages on a single AMQP link.
 type Sender struct {
-	link *link
+	link
```

Comment:
I will point out that embedding `link` like this means that we have to be very careful not to add any exported fields/methods on `link` (I was on the fence about doing it this way for just this reason).  It might be better to make this a field instead.

Reply:
I went ahead and made this change as it removes the possibility for inadvertent changes to public surface area and sets us up for future pimpl refactoring.